src/main/java/EchoClient.java

```diff
@@ -0,0 +1,42 @@
+import java.io.IOException;
+import java.net.StandardProtocolFamily;
+import java.nio.ByteBuffer;
+import java.nio.channels.spi.SelectorProvider;
+import java.nio.charset.StandardCharsets;
+
+public class EchoClient {
```

Comment:
It is still not reusable. Consider to use it as API

Reply:
Created EchoClient component